### PR TITLE
Lint 

### LINKED ISSUE #141: Eslint update
Update the eslint configurations.

### PR DESCRIPTION
**Pull request by Bashamega**
closes #141 
I have upgraded lint and modified the code to be compatible with lint

### CODE REVIEW

**Review by LamboCreeper — COMMENTED** [review]
It doesn't look like you're actually extending `eslint-config-codesupport` anywhere, and our previous config is being completely overwritten?

**Comment by Bashamega** [comment]
Sorry I forgot. Do you mean something like this: https://eslint.org/docs/latest/extend/shareable-configs

**Comment by Bashamega** [comment]
@LamboCreeper

**Comment by LamboCreeper** [comment]
Apologies, I must have missed this. Yes, we have our own package already: https://github.com/codesupport/eslint-config-codesupport

Central code style configuration should be here, so that it can be easily shared across projects.

**Comment by Bashamega** [comment]
Hello @LamboCreeper :)

Sorry, for the delay. I have updated the config. And everything should be good

File: eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export default [
+  {files: ["**/*.{js,mjs,cjs,ts,jsx,tsx}"]},
+  { languageOptions: { parserOptions: { ecmaFeatures: { jsx: true } } } },
+  {files: ["**/*.js"], languageOptions: {sourceType: "script"}},
+  {languageOptions: { globals: globals.browser }},
+  pluginJs.configs.recommended,
+  ...tseslint.configs.recommended,
+  pluginReactConfig,
+];
```
Comment (LamboCreeper) [inline eslint.config.js:15]: Ironically, this file isn't being linted:
```suggestion
import globals from "globals";
import pluginJs from "@eslint/js";
import tseslint from "typescript-eslint";
import pluginReactConfig from "eslint-plugin-react/configs/recommended.js";

export default [
  { files: ["**/*.{js,mjs,cjs,ts,jsx,tsx}"] },
  { languageOptions: { parserOptions: { ecmaFeatures: { jsx: true } } } },
  { files: ["**/*.js"], languageOptions: { sourceType: "script" } },
  { languageOptions: { globals: globals.browser } },
  pluginJs.configs.recommended,
  ...tseslint.configs.recommended,
  pluginReactConfig,
];
```

**Review by LamboCreeper — CHANGES_REQUESTED** [review]

**Review by LamboCreeper — APPROVED** [review]

**Pull request merged**